docs/prometheus-metrics.md

```diff
@@ -0,0 +1,58 @@
+# Prometheus Metrics Coverage
+
+We plan to collect a rich set of metrics in kubeflow/common's `JobController` using [Prometheus](https://prometheus.io/).
+The goal is to report generic metrics (e.g. metrics related to pods/jobs/services) during the lifecycle of `JobController` so that:
+
+* Other operators built on top of it will automatically report Prometheus metrics without additional efforts;
+* It is easier for users of Kubeflow distributed training operators to monitor operator performance and behaviors using consistent set of metrics for different distributed training operators.
+
+This document outlines the list of Prometheus metrics we plan to cover in `JobController`.
+
+## Pod Metrics
+
+The following metrics related to the lifecycle of pods will be reported:
+
+* The total number of created pods
```

Comment:
I think we can list the name of the metrics here.

Reply:
Yeah. That would be great. We can use them directly in implementation.

Reply:
Good suggestion. Added.